### PR TITLE
BookStore : Add reducers and actions

### PR DESCRIPTION
**Pull request by bhushan354**
In this PR I have added following things :

- Added Redux Toolkit .
- Created a directory that will contain all  Redux logic (/src/redux).
- Configured a Redux store (/src/redux/store.js)
- Defined a slice of state for categories.
- Defined a slice of state for books.

Thank you ,
Bhushan.

### CODE REVIEW

File: src/store.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* eslint-disable import/prefer-default-export */
+
+import { configureStore } from '@reduxjs/toolkit';
+
+export const store = configureStore({
+  reducer: {},
+});
```
Comment (ShadyShawkat) [inline src/store.js]: - [ ] [OPTIONAL] Please remove this file as you already have a `store` file for the redux.

File: src/redux/store.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable */
+import { configureStore } from '@reduxjs/toolkit';
+import categoriesReducer from './categories/categoriesSlice';
+import bookReducer from './books/booksSlice.js';
+/* eslint-enable */
```
Comment (ShadyShawkat) [inline src/redux/store.js]: - [ ] Kindly remove `eslint-disable` as it defies te purpose of using inters in the first place.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @bhushan354, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/store.js (hunk 2 above)
Comment (bhushan354) [reply src/redux/store.js]: hello sir @ShadyShawkat if i enable eslint there it asks to add file extension of import statement then when i add the .js extension for file then eslint says unexpected file extension

**Comment by bhushan354** [comment]
Hello sir , if i enable eslint there it asks to add file extension of import statement then when i add the .js extension for file then eslint says unexpected file extension :
Screenshot from 2023-09-29 19-52-49

File: src/redux/store.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable */
+import { configureStore } from '@reduxjs/toolkit';
+import categoriesReducer from './categories/categoriesSlice';
+import bookReducer from './books/bookSlice';
+/* eslint-enable */
+
```
Comment (skyv26) [inline src/redux/store.js]: - [ ] Please always implement the things instructed by your code reviewer. I can see that after getting the review, you still haven't removed the `eslinter` disable and enable comment. As I tried at my end, there is no error after removing it. So please remove these comments in order to get project approval.

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @bhushan354 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## STATUS: REQUIRE CHANGES 🔴 




## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @bhushan354 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on fixing the Linters problem 💯 

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**